src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt

```diff
@@ -0,0 +1,110 @@
+package at.hannibal2.skyhanni.features.nether
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.config.features.crimsonisle.SulphurSkitterBoxConfig
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.jsonobjects.repo.ItemsJson
+import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.events.RepositoryReloadEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.LocationUtils
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceToPlayer
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.editCopy
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RenderUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.drawDynamicText
+import at.hannibal2.skyhanni.utils.RenderUtils.expandBlock
+import at.hannibal2.skyhanni.utils.SpecialColour
+import at.hannibal2.skyhanni.utils.toLorenzVec
+import net.minecraft.client.Minecraft
+import net.minecraft.init.Blocks
+import net.minecraft.util.AxisAlignedBB
+import net.minecraft.util.BlockPos
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+
+class SulphurSkitterBox {
+
+    private val config get() = SkyHanniMod.feature.crimsonIsle.sulphurSkitterBoxConfig
+    private val rodsList = mutableListOf<String>()
+    private var blocksList = listOf<LorenzVec>()
+    private val radius = 8
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+        if (!event.repeatSeconds(1)) return
+        if (InventoryUtils.itemInHandId.asString() !in rodsList && config.onlyWithRods) return
+        blocksList = emptyList()
+        val from = LocationUtils.playerLocation().add(-20, -20, -20).toBlockPos()
+        val to = LocationUtils.playerLocation().add(20, 20, 20).toBlockPos()
+        for (blockPos in BlockPos.getAllInBox(from, to)) {
+            val b = Minecraft.getMinecraft().theWorld.getBlockState(blockPos).block
+            if (b == Blocks.sponge && blockPos.toLorenzVec().distanceToPlayer() <= 15) {
+                blocksList = blocksList.editCopy { add(blockPos.toLorenzVec()) }
+            }
+        }
```

Comment:
There is no need to `.editCopy` for each new block. Just store the changes all at once at the end.

```suggestion
        val from = LocationUtils.playerLocation().add(-20, -20, -20).toBlockPos()
        val to = LocationUtils.playerLocation().add(20, 20, 20).toBlockPos()
        blocksList = BlockPos.getAllInBox(from, to).filter {
            val b = Minecraft.getMinecraft().theWorld.getBlockState(blockPos).block
            b == Blocks.sponge && blockPos.toLorenzVec().distanceToPlayer() <= 15
        }   
```